src/js/events/ChronosActions.js

```diff
@@ -17,11 +18,17 @@ const ChronosActions = {
           url: `${Config.rootUrl}/chronos/jobs`,
           data: [{name: 'embed', value: 'activeJobs'}],
           success: function (response) {
-            AppDispatcher.handleServerAction({
-              type: REQUEST_CHRONOS_JOBS_SUCCESS,
-              data: response
-            });
-            resolve();
+            let data;
+            try {
+              data = ChronosUtil.parseJobs(response);
+              AppDispatcher.handleServerAction({
+                type: REQUEST_CHRONOS_JOBS_SUCCESS,
+                data
+              });
+              resolve();
+            } catch (error) {
+              this.error(error);
+            }
```

Comment:
So in the description, I tried to explain that this using a more reducer-like approach to the data-parsing, meaning that it parses up in front, instead of when we build out the struct. The latter approach could become CPU heavy. The same change will be done for the ServiceTree.